File: ios/Mobius-Common/Model/Item.swift
import Foundation

public struct Item: Decodable {
    let itemID: String
    let name: String
    let quantity: Int
    let pricePerUnit: Double
    let currency: String
    let imageURL: String

    enum CodingKeys: String, CodingKey {
        case itemID = "item_id"
        case name
        case quantity
        case pricePerUnit = "price_per_unit"
        case currency
        case imageURL = "image_url"
    }
}

File: Mobius-Demo-Flutter/lib/feature/order/data/model/item.dart
import 'package:meta/meta.dart';

import '../../domain/entity/entity.dart';

@immutable
class ItemModel extends Item {
  const ItemModel({
    required super.itemID,
    required super.name,
    required super.quantity,
    required super.pricePerUnit,
    required super.currency,
    required super.imageURL,
  });

  factory ItemModel.fromMap(Map<String, Object?> map) {
    return ItemModel(
      itemID: map['item_id'] as String,
      name: map['name'] as String,
      quantity: map['quantity'] as int,
      pricePerUnit: (map['price_per_unit'] as num).toDouble(),
      currency: map['currency'] as String,
      imageURL: map['image_url'] as String,
    );
  }

  Map<String, Object?> toMap() {
    return {
      'item_id': itemID,
      'name': name,
      'quantity': quantity,
      'price_per_unit': pricePerUnit,
      'currency': currency,
      'image_url': imageURL,
    };
  }
}
